crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -65,13 +64,14 @@ pub(super) fn configure_dsn(
 
     // Metrics
     let mut metrics_registry = Registry::default();
-    let metrics = initialize_metrics.then(|| Metrics::new(&mut metrics_registry));
+    let dsn_metric_registry = initialize_metrics.then_some(&mut metrics_registry);
 
     let default_config = Config::new(
         protocol_prefix,
         keypair,
         piece_cache.clone(),
         Some(PeerInfoProvider::new_farmer()),
+        dsn_metric_registry,
```

Comment:
I would just do this instead:
```suggestion
    let default_config = Config::new(
        protocol_prefix,
        keypair,
        piece_cache.clone(),
        Some(PeerInfoProvider::new_farmer()),
        initialize_metrics.then_some(&mut metrics_registry),
```

Also we don't need to call things `dsn_X` if they are not specific to DSN.

In fact I'd refactor this to take `Option<&mut Registry>` instead of `initialize_metrics` because we'll soon have other metrics on the farmer. This is what I meant by saying it should be an implementation detail.